### ISSUE TITLE
move the resolover cache handling from fail over to a new layer atop resolver

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#1087]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/1083
- Created at 2011-11-11 10:17:21 by [jhrozek](https://pagure.io/user/jhrozek)
- Closed at 2020-03-24 14:12:48 as wontfix
- Assigned to nobody
---

The asynchronous resolver currently always performs online resolution and returns TTL of the result. The decision on whether to reuse the cached result or resolve online again is done in the layer above the resolver - the fail over code.

We should move the cache evaluation into the resolver itself or create a very thin layer on top of the resolver. This would help other consumers of the resolver such as the upcoming referral chasing code.

### Comments

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2011-11-17 14:53:57](https://pagure.io/SSSD/sssd/issue/1083#228089)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.7.0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2011-11-21 10:37:13](https://pagure.io/SSSD/sssd/issue/1083#228090)**

Fields changed

owner: somebody => jhrozek

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2011-11-21 10:37:19](https://pagure.io/SSSD/sssd/issue/1083#228091)**

Fields changed

status: new => assigned

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2011-11-21 14:13:36](https://pagure.io/SSSD/sssd/issue/1083#228092)**

Fields changed

priority: major => blocker

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2011-12-08 15:09:22](https://pagure.io/SSSD/sssd/issue/1083#228093)**

Fields changed

milestone: SSSD 1.7.0 => Referrals

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2012-01-19 02:14:28](https://pagure.io/SSSD/sssd/issue/1083#228094)**

Fields changed

rhbz:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2013-01-03 16:17:03](https://pagure.io/SSSD/sssd/issue/1083#228095)**

Fields changed

blockedby:  => 
blocking:  => 
design:  => 
design_review:  => 0
feature_milestone:  => 
fedora_test_page:  => 
priority: blocker => critical
selected:  =>

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2013-03-27 14:58:50](https://pagure.io/SSSD/sssd/issue/1083#228096)**

Fields changed

milestone: SSSD Referrals Feature => NEEDS_TRIAGE
review:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2013-03-27 14:59:30](https://pagure.io/SSSD/sssd/issue/1083#228097)**

Fields changed

priority: critical => major
summary: move the resolover cache handling from fail over to resolver itself => move the resolover cache handling from fail over to a new layer atop resolver

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2013-03-28 14:14:35](https://pagure.io/SSSD/sssd/issue/1083#228098)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.11 beta

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-07-30 12:23:09](https://pagure.io/SSSD/sssd/issue/1083#228099)**

Fields changed

blocking:  => #2394
changelog:  =>

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-09-26 18:49:56](https://pagure.io/SSSD/sssd/issue/1083#228100)**

Fields changed

mark:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2015-02-10 14:03:44](https://pagure.io/SSSD/sssd/issue/1083#228101)**

This is definitely the right thing to do, but we won't touch the resolver code unless the changes are required by the 1.13 failover/resolver RFEs we plan.

milestone: SSSD 1.13 beta => SSSD 1.13 backlog
owner: jhrozek => somebody
status: assigned => new
type: defect => enhancement

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2015-02-12 20:12:46](https://pagure.io/SSSD/sssd/issue/1083#228102)**

Mass-moving tickets not planned for the 1.13 release to 1.14

milestone: SSSD 1.13 backlog => SSSD 1.14 beta

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-02-16 12:28:00](https://pagure.io/SSSD/sssd/issue/1083#228103)**

Refactoring the resolver and failover is out of scope of 1.14.

milestone: SSSD 1.14 beta => SSSD 1.15 beta
sensitive:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-02-24 14:37:30](https://pagure.io/SSSD/sssd/issue/1083#228104)**

**Metadata Update from @jhrozek**:
- Issue marked as blocked by: #2394
- Issue set to the milestone: SSSD Future releases (no date set yet)

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-13 14:31:01](https://pagure.io/SSSD/sssd/issue/1083#633203)**

**Metadata Update from @thalman**:
- Custom field blocking reset (from #2394)
- Custom field design_review reset (from 0)
- Custom field mark reset (from 0)
- Custom field patch reset (from 0)
- Custom field review reset (from 0)
- Custom field sensitive reset (from 0)
- Custom field testsupdated reset (from 0)
- Issue close_status updated to: None
- Issue tagged with: Canditate to close

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-24 14:12:47](https://pagure.io/SSSD/sssd/issue/1083#636714)**

Thank you for taking time to submit this request for SSSD. Unfortunately this issue was not given priority and the team lacks the capacity to work on it at this time.

Given that we are unable to fulfill this request I am closing the issue as wontfix.

If the issue still persist on recent SSSD you can request re-consideration of this decision by reopening this issue. Please provide additional technical details about its importance to you.

Thank you for understanding.

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-24 14:12:48](https://pagure.io/SSSD/sssd/issue/1083#636715)**

**Metadata Update from @pbrezina**:
- Issue close_status updated to: wontfix
- Issue status updated to: Closed (was: Open)
